api/lagoon/v1beta2/lagoonbuild_helpers.go

```diff
@@ -653,11 +690,239 @@ func SortQueuedNamespaceBuilds(namespace string, pendingBuilds []string) ([]Cach
+	cl client.Client,
+	namespace *corev1.Namespace,
+	lagoonBuild LagoonBuild,
+	opLog logr.Logger,
+	controllerNamespace, namespacePrefix string,
+	randomNamespacePrefix bool) (bool, error) {
+	// parse the project/env through the project pattern, or use the default
+	ns := helpers.GenerateNamespaceName(
+		lagoonBuild.Spec.Project.NamespacePattern, // the namespace pattern or `openshiftProjectPattern` from Lagoon is never received by the controller
+		lagoonBuild.Spec.Project.Environment,
+		lagoonBuild.Spec.Project.Name,
+		namespacePrefix,
+		controllerNamespace,
+		randomNamespacePrefix,
+	)
+	nsLabels := map[string]string{
+		"lagoon.sh/project":         lagoonBuild.Spec.Project.Name,
+		"lagoon.sh/environment":     lagoonBuild.Spec.Project.Environment,
+		"lagoon.sh/environmentType": lagoonBuild.Spec.Project.EnvironmentType,
+		"lagoon.sh/controller":      controllerNamespace,
+	}
+	if lagoonBuild.Spec.Project.Organization != nil {
+		nsLabels["organization.lagoon.sh/id"] = fmt.Sprintf("%d", *lagoonBuild.Spec.Project.Organization.ID)
+		nsLabels["organization.lagoon.sh/name"] = lagoonBuild.Spec.Project.Organization.Name
+	}
+	if lagoonBuild.Spec.Project.ID != nil {
+		nsLabels["lagoon.sh/projectId"] = fmt.Sprintf("%d", *lagoonBuild.Spec.Project.ID)
+	}
+	if lagoonBuild.Spec.Project.EnvironmentID != nil {
+		nsLabels["lagoon.sh/environmentId"] = fmt.Sprintf("%d", *lagoonBuild.Spec.Project.EnvironmentID)
+	}
+	// set the auto idling values if they are defined
+	if lagoonBuild.Spec.Project.EnvironmentIdling != nil {
+		// eventually deprecate 'lagoon.sh/environmentAutoIdle' for 'lagoon.sh/environmentIdlingEnabled'
+		nsLabels["lagoon.sh/environmentAutoIdle"] = fmt.Sprintf("%d", *lagoonBuild.Spec.Project.EnvironmentIdling)
+		if *lagoonBuild.Spec.Project.EnvironmentIdling == 1 {
+			nsLabels["lagoon.sh/environmentIdlingEnabled"] = "true"
+		} else {
+			nsLabels["lagoon.sh/environmentIdlingEnabled"] = "false"
+		}
+	}
+	if lagoonBuild.Spec.Project.ProjectIdling != nil {
+		// eventually deprecate 'lagoon.sh/projectAutoIdle' for 'lagoon.sh/projectIdlingEnabled'
+		nsLabels["lagoon.sh/projectAutoIdle"] = fmt.Sprintf("%d", *lagoonBuild.Spec.Project.ProjectIdling)
+		if *lagoonBuild.Spec.Project.ProjectIdling == 1 {
+			nsLabels["lagoon.sh/projectIdlingEnabled"] = "true"
+		} else {
+			nsLabels["lagoon.sh/projectIdlingEnabled"] = "false"
+		}
+	}
+	if lagoonBuild.Spec.Project.StorageCalculator != nil {
+		if *lagoonBuild.Spec.Project.StorageCalculator == 1 {
+			nsLabels["lagoon.sh/storageCalculatorEnabled"] = "true"
+		} else {
+			nsLabels["lagoon.sh/storageCalculatorEnabled"] = "false"
+		}
+	}
+	// add the required lagoon labels to the namespace when creating
+	namespace.ObjectMeta = metav1.ObjectMeta{
+		Name:   ns,
+		Labels: nsLabels,
+	}
+
+	// if kubernetes, just create it if it doesn't exist
+	if err := cl.Get(ctx, types.NamespacedName{Name: ns}, namespace); err != nil {
+		if err := cl.Create(ctx, namespace); err != nil {
```

Comment:
I wonder if there isn't any utility to checking for other kind of error types and not simply assuming that the error is that the ns doesn't exist?
I mean, if something's failing, it'll _probably_ also fail on the create as well.

Reply:
There is an api error wrapper that can check if the return error is not found, I'll fix it up to do that so that other errors are captured